test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +92,7 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(command.isMutating()) {assertEquals(addressBook, saveFile.load());}
```

Comment:
You should test the else branch where the command is not mutating. You can think about how to test that.